instruments/tektronix/tekdpo70000.py

```diff
@@ -190,12 +190,8 @@ def read_waveform(self, bin_format=True):
                 )
                 self._parent.sendcmd("CURV?")
                 raw = self._parent.binblockread(n_bytes, fmt=dtype)
-                # Clear the queue by trying to read.
-                # FIXME: this is a hack-y way of doing so.
-                if hasattr(self._parent._file, 'flush_input'):
-                    self._parent._file.flush_input()
-                else:
-                    self._parent._file.read()
+                # Clear the queue by reading the end of line character
+                self._parent._file.read_raw(1)
```

Comment:
Here is the simplified way of clearing the queue, used this already in PR #253. The manual clearly states the the waveform transfer is succeeded by a `<newline>` character. Reading this character is required to flush the input. 

This takes care of the earlier issue in which the tests failed since the `loopback_communicator` does not flush the input properly and simply passes that statement.

Reply:
If the instrument says it always sends this character, then we should read it in. I'm good with this.

Reply:
To keep a record - the [manual](https://www.tek.com/oscilloscope/dpo70000-mso70000-manual/dpo70000sx-mso-dpo70000dx-mso-dpo70000c-dpo7000c-mso5000-b-1) states on page 2-281:

> `<Block>`is the waveform data in binary format. The waveform is formatted as:`#<x><yyy><data><newline>`, where:
  [...]
  `<newline>` is a single byte new line character at the end of the data.